core/src/subgraph/registrar.rs

```diff
@@ -651,96 +728,182 @@ fn create_subgraph_version(
+                        Some(subgraph_version_data.version_entity_id),
+                    ));
+                }
+                SubgraphVersionSwitchingMode::Synced => {
+                    // Set the new version as pending, unless there is no fully synced
+                    // current versions
+                    if subgraph_version_data.current_is_synced {
+                        ops.extend(SubgraphEntity::update_pending_version_operations(
+                            &subgraph_version_data.subgraph_entity_id,
+                            Some(subgraph_version_data.version_entity_id),
+                        ));
+                    } else {
+                        ops.extend(SubgraphEntity::update_pending_version_operations(
+                            &subgraph_version_data.subgraph_entity_id,
+                            None,
+                        ));
+                        ops.extend(SubgraphEntity::update_current_version_operations(
+                            &subgraph_version_data.subgraph_entity_id,
+                            Some(subgraph_version_data.version_entity_id),
+                        ));
+                    }
+                }
             }
-        }
-    }
 
-    // Commit entity ops
-    let manifest_id = manifest.id.to_string();
-    if deployment_exists {
-        store.apply_metadata_operations(ops)?
-    } else {
-        store.create_subgraph_deployment(&manifest.schema, ops)?;
-    }
-
-    debug!(
-        logger,
-        "Wrote new subgraph version to store";
-        "subgraph_name" => name.to_string(),
-        "subgraph_hash" => manifest_id,
-    );
+            // Check if deployment exists
+            store
+                .get(SubgraphDeploymentEntity::key(manifest_id.clone()))
+                .map_err(|e| SubgraphRegistrarError::QueryExecutionError(e))
+                .map(|a| (logger, ops, a.is_some()))
+        })
+        .and_then(move |(logger, mut ops, deployment_exists)| {
+            ops.push(MetadataOperation::AbortUnless {
+                description: "Subgraph deployment entity must continue to exist/not exist"
+                    .to_owned(),
+                query: SubgraphDeploymentEntity::query()
+                    .filter(EntityFilter::new_equal("id", manifest.id.to_string())),
+                entity_ids: if deployment_exists {
+                    vec![manifest.id.to_string()]
+                } else {
+                    vec![]
+                },
+            });
 
-    Ok(())
+            resolve_subgraph_chain_blocks(
+                manifest.clone(),
+                chain_store.clone(),
+                ethereum_adapter.clone(),
+                &logger.clone(),
+            )
+            .and_then(move |(chain_head_block, earliest_block)| {
+                info!(
+                    logger,
+                    "Set the earliest available Ethereum block, start_block: {}",
+                    earliest_block.number + 1
```

Comment:
This line confuses me. I'm assuming `earliest_block` is the smallest start block across all data sources. Shouldn't we then start `earliest_block.number` by setting the "latest block" to `earliest_block.number - 1` when creating the deployment?

Reply:
Or is this completely unrelated to data source start blocks?